modules/gapi/test/serialization/serialization.cpp

```diff
@@ -0,0 +1,1292 @@
+    gimpl::passes::dumpDotToFile(pass_ctx, "graph.dot");
+}
+
+/////////////////////////////////////////////////////////////////////////////////////
+//Graph dump operators
+template<typename _Tp> static inline
+SerializationStream&  operator << (SerializationStream& os, /*const*/ _Tp value)
+{
+    os << value;
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, Kernel k)
+{
+    CV_LOG_INFO(NULL, "k.name  " << k.name.c_str());
+    CV_LOG_INFO(NULL, "k.tag  " << k.tag.c_str());
+    os << k.name;
+    os << k.tag;
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, std::string str)
+{
+    uint str_size = (uint)str.size();
+    os << str_size;
+    size_t numAtoms = str_size % sizeof(uint) ==
+        0 ? (str_size / sizeof(uint)) : (str_size / sizeof(uint)) + 1;
+    uint* hton_buff = (uint*)malloc(numAtoms * sizeof(uint));
+    memcpy(hton_buff, str.c_str(), str_size);
+    for (uint a = 0; a < numAtoms; a++)
+    {
+        os << hton_buff[a];
+    }
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, std::vector<int> ints)
+{
+    os << (uint)ints.size();
+    for (auto & element : ints)
+    {
+        os << element;
+    }
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, std::vector<RcDesc> descs)
+{
+    os << (uint)descs.size();
+    for (auto & element : descs)
+    {
+        os << element;
+    }
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, RcDesc desc)
+{
+    os << desc.id;
+    os << (int)desc.shape;
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, std::vector<double> doubles)
+{
+    os << (uint)doubles.size();
+    for (auto & element : doubles)
+    {
+        os << element;
+    }
+    return os;
+}
+
+SerializationStream&  operator << (SerializationStream& os, double double_val)
+{
+    uint element_tmp[2];
+    memcpy(&element_tmp, &double_val, 2 * sizeof(uint));
+    os << element_tmp[0];
+    os << element_tmp[1];
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, std::vector<cv::Size> cvsizes)
+{
+    os << (uint)cvsizes.size();
+    for (auto & element : cvsizes)
+    {
+        os << element;
+    }
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, cv::Size cvsize)
+{
+    os << cvsize.width;
+    os << cvsize.height;
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, std::vector<bool> bools)
+{
+    uint bools_size = (uint)bools.size();
+    os << bools_size;
+    for (uint j = 0; j < bools_size; j++)
+    {
+        int bool_val_int;
+        bool_val_int = bools[j] ? 1 : 0;
+        os << bool_val_int;
+        //os << bools[j];
+    }
+    return os;
+}
+
+SerializationStream&  operator << (SerializationStream& os, bool bool_val)
+{
+    int bool_val_int;
+    bool_val_int = bool_val ? 1 : 0;
+    os << bool_val_int;
+    return os;
+}
```

Comment:
bool is an atom.

Reply:
The same comment as for atom. No uint is an atom.

Reply:
No, bool is an atom.